lib/mix/lib/mix/dep/lock.ex

```diff
@@ -32,48 +32,79 @@ defmodule Mix.Dep.Lock do
   """
   @spec read() :: map
   def read() do
-    case File.read(lockfile()) do
-      {:ok, info} ->
-        assert_no_merge_conflicts_in_lockfile(lockfile(), info)
-
-        case Code.eval_string(info, [], file: lockfile()) do
-          {lock, _binding} when is_map(lock) -> lock
-          {_, _binding} -> %{}
-        end
-
+    lockfile = lockfile()
+    case File.read(lockfile) do
+      {:ok, content} ->
+        read_lock(content, lockfile)
       {:error, _} ->
         %{}
     end
   end
 
+  @start "<<<<<<<"
+  @separator "======="
+  @finish ">>>>>>>"
+  @start_regex ~r/#{@start} .+\n/
+  @finish_regex ~r/#{@finish} .+\n/
+  @ancestor_regex ~r/\|{7} .+\n[\w\W\s]+\n#{@separator}/
+  @conflict_regex ~r/#{@start} .+\n([\w\W\s]+)\n#{@separator}\n([\w\W\s]+)\n#{@finish} .+\n/
+
+  defp read_lock(content, lockfile) do
+    if Regex.match?(@conflict_regex, content) do
+      resolve_merge_conflicts(content, lockfile)
+    else
+      eval_content(content, lockfile)
+    end
+  end
+
+  defp resolve_merge_conflicts(content, lockfile) do
+    new_map =
+      content
+      |> String.replace(@ancestor_regex, @separator)
+      |> String.replace(@start_regex, "")
+      |> String.replace(@separator, "")
+      |> String.replace(@finish_regex, "")
```

Comment:
This is not the most performant version, I know, but as this is not being called that much, I preferred to keep the code easy to read.